### PR TITLE
Add database and key vault

### PR DESCRIPTION
**Pull request by WillTaylor22**

### CODE REVIEW

File: db.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+module "tt-database" {
+  source                = "git@github.com:hmcts/cnp-module-postgres?ref=master"
```
Comment (timja) [inline db.tf]:
```suggestion
  source                = "git@github.com:hmcts/cnp-module-postgres?ref=postgresql_tf"
```

File: db.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
+module "tt-database" {
```
Comment (timja) [inline db.tf:1]:
```suggestion
data "azurerm_subnet" "postgres" {
  name                 = "iaas"
  resource_group_name  = "ss-${var.env}-network-rg"
  virtual_network_name = "ss-${var.env}-vnet"
}

module "tt-database" {
```

File: db.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  product               = var.product
+  component             = var.db_component_name
+  location              = var.location
+  env                   = var.env
+  postgresql_user       = var.db_postgresql_user
+  database_name         = var.db_name
+  postgresql_version    = var.db_version
+  common_tags           = var.common_tags
+  subscription          = var.subscription
+  storage_mb            = var.db_storage_mb
```
Comment (timja) [inline db.tf:12]:
```suggestion
  storage_mb            = var.db_storage_mb
  business_area        = "SDS"
  key_vault_rg       = "genesis-rg"
  key_vault_name     = "dtssharedservices${var.env}kv"
  subnet_id          = data.azurerm_subnet.postgres.id
```

**Review by timja — COMMENTED** [review]
you'll need all this as well

File: stg.tfvars [hunk 4]
```diff
@@ -0,0 +1,4 @@
+"env" = "stg"
```
Comment (timja) [inline stg.tfvars]: this is set via the pipeline don't add it here

File: stg.tfvars [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+"db_name" = "tt_stg"
+"db_storage_mb" = "5120"
```
Comment (timja) [inline stg.tfvars]: leave a sensible default and only override it where you need

File: stg.tfvars [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+"subscription" = "DTS-SHAREDSERVICES-STG"
```
Comment (timja) [inline stg.tfvars]: I doubt the variable is supposed to map to this it's probably something like stg, doesn't need to be committed, you can have a `local.tfvars` if you want to make it easier to run locally

File: stg.tfvars [hunk 7]
```diff
@@ -0,0 +1,4 @@
+"env" = "stg"
+"db_name" = "tt_stg"
```
Comment (timja) [inline stg.tfvars]: doesn't need to be a variable just use something like `var.product` in your main.tf

File: prod.tfvars [hunk 8]
```diff
@@ -0,0 +1,4 @@
+"env" = "prod"
+"db_name" = "tt_prod"
+"db_storage_mb" = "20480"
```
Comment (timja) [inline prod.tfvars:1]: unless you know you need this best to just leave the default

File: output.tf [hunk 9]
```diff
@@ -0,0 +1,7 @@
+output "vaultName" {
+  value = module.tt-key-vault.key_vault_name
+}
+
+output "vaultUri" {
+  value = module.tt-key-vault.key_vault_uri
```
Comment (timja) [inline output.tf]: not needed

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,15 @@
+terraform {
+  required_version = "1.23.0"
```
Comment (timja) [inline main.tf]: 1.23.0 isn't a thing as far as i can tell
```suggestion
  required_version = "1.1.2"
```

**Review by timja — CHANGES_REQUESTED** [review]

**Review by timja — CHANGES_REQUESTED** [review]
Also missing a `Jenkinsfile_CNP` e.g. see https://github.com/hmcts/sds-toffee-shared-infrastructure/blob/master/Jenkinsfile_CNP

File: prod.tfvars (hunk 8 above)
Comment (WillTaylor22) [reply prod.tfvars:1]: The previous DB we are migrating from was also that size, so it does need to match

File: variables-defaults.tf [hunk 11]
```diff
@@ -0,0 +1,66 @@
+# Variables with default values or values specified in an {env}.tfvars file
+
+variable "product" {
+  type = "string"
+  default = "tax-tribunals"
```
Comment (timja) [inline variables-defaults.tf]: generally better to leave it out this is set via Jenkins
```suggestion
```

File: variables-defaults.tf [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+variable "product_name_short" {
+  type = "string"
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
```

File: variables-defaults.tf [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+variable "common_tags" {
+  type = "map"
+  default = {
+    "tag_service" = "tt",
+    "tag_project_name" = "pet-devops",
+    "tag_cost_center" = "iCode-Drone",
+    "tag_app_operations_owner" = "pet-devops",
+    "tag_system_owner" = "pet-devops",
+    "tag_budget_owner" = "pet-devops"
+  }
```
Comment (timja) [inline variables-defaults.tf]: this doesn't map to the right tags at all, Jenkins will setup the right one
```suggestion
```

File: variables-defaults.tf [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+variable "subscription" {
+  type = "string"
+  default = "DTS-SHAREDSERVICES-${upper(var.env)}"
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
```

File: variables-defaults.tf [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+# Database
+
+variable "db_name" {
+  type = "string"
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
```

File: variables-defaults.tf [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+variable "db_storage_mb" {
+  type = "string"
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
```

File: variables-defaults.tf [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+  default = "5120"
+}
+
+variable "db_component_name" {
+  type = "string"
+  default = "postgres-database"
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
  default = "service"
```

File: variables-defaults.tf [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+variable "db_name" {
+  type = "string"
+}
+
+variable "db_storage_mb" {
+  type = "string"
+  default = "5120"
+}
+
+variable "db_component_name" {
```
Comment (timja) [inline variables-defaults.tf]:
```suggestion
variable "component" {
```

**Review by timja — COMMENTED** [review]
this should clean it up more

File: key-vault.tf [hunk 19]
```diff
@@ -0,0 +1,12 @@
+module "tt-key-vault" {
+  source                  = "git@github.com:hmcts/cnp-module-key-vault?ref=master"
+  name                    = "${product}-${env}"
```
Comment (timja) [inline key-vault.tf]: not needed if you have a short enough name which yours should be, especially given you just put the default in:
```suggestion
```

**Review by timja — APPROVED** [review]

**Pull request merged**